consumer.go

```diff
@@ -628,6 +654,9 @@ func (child *partitionConsumer) parseResponse(response *FetchResponse) ([]*Consu
 		// We got no messages. If we got a trailing one then we need to ask for more data.
 		// Otherwise we just poll again and wait for one to be produced...
 		if partialTrailingMessage {
+			if child.fetchStatsEnabled() && metricRegistry != nil {
+				getOrRegisterPartitionCounter("consumer-partial-fetches", child.topic, child.partition, metricRegistry).Inc(1)
```

Comment:
It seems like adding this function would deduplicate some code:

```
func (child *partitionConsumer) count(name string) {
    reg := child.conf.MetricRegistry
    if reg != nil && child.fetchStatsEnabled() {
         getOrRegisterPartitionCounter(name, child.topic, child.partition, reg).Inc(1)
    }
}
```

Then this line becomes `child.count("consumer-partial-fetches")`, and so on for the similar two above.

You could also consider moving all of the logic in from `metrics`:

func (child *partitionConsumer) count(name string) {
    r := child.conf.MetricRegistry
    if r != nil && child.fetchStatsEnabled() {
         metrics.GetOrRegisterCounter(getMetricNameForPartition(name, child.topic, child.partition), r)
    }
}

And likewise for `Meter` and `Histogram`.